2-mandatory/3-shopping-list.js

```diff
@@ -19,35 +19,54 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
-}
+  let shoppingList = {};
+  shoppingListItems = shoppingList["items"]
+  shoppingList.name = recipe.name;
+  for (let element of recipe.ingredients) {
+    if (!pantry.fridgeContents.includes(element)) {
+      shoppingListItems = element
+    } else if (!pantry.cupboardContents.includes(element)) {
+      shoppingListItems = element
+    }
+  }
+  return shoppingList   
+  }
+let recipe1 = {
+    name: "pancakes",
+    ingredients: ["flour", "salt", "milk", "eggs", "vegetable oil"],
+  };
+console.log(createShoppingList(recipe1));
```

Comment:
Consider: Also here a .forEach can be useful.
function createShoppingList(recipe) {
  // write code here

  let shoppingList = {
    items: [],
  }
  recipe.ingredients.forEach((ingredient) => {
    if (!pantry.fridgeContents.includes(ingredient) && !pantry.cupboardContents.includes(ingredient)){
      shoppingList.items.push(ingredient)
    }
  });
  shoppingList.name = recipe.name
  return shoppingList
}